tests/pipeline/controlflowanalysis/test_variable_name_generation.py

```diff
@@ -108,52 +109,72 @@ def test_default_notation_1():
 
 
 def test_custom_type():
     true_value = LogicCondition.initialize_true(LogicCondition.generate_new_context())
-    ast = AbstractSyntaxTree(CodeNode(Assignment(var := Variable("var_0", CustomType("size_t", 64)), Constant(0)), true_value), {})
+    ast = AbstractSyntaxTree(CodeNode([assignment := Assignment(Variable("var_0", CustomType("size_t", 64)), Constant(0))], true_value), {})
     _run_vng(ast, _generate_options())
-    assert var._name == "Var0"
+    assert assignment.destination.name == "unkVar0"
 
 
 def test_bninja_invalid_type():
     true_value = LogicCondition.initialize_true(LogicCondition.generate_new_context())
-    ast = AbstractSyntaxTree(CodeNode(Assignment(var := Variable("var_0", Integer(104, True)), Constant(0)), true_value), {})
+    ast = AbstractSyntaxTree(CodeNode([assignment := Assignment(Variable("var_0", Integer(104, True)), Constant(0))], true_value), {})
     _run_vng(ast, _generate_options())
-    assert var._name == "unkVar0"
-
-
-def test_tmp_variable():
-    true_value = LogicCondition.initialize_true(LogicCondition.generate_new_context())
-    ast = AbstractSyntaxTree(CodeNode(Assignment(var := Variable("tmp_42", Float(64)), Constant(0)), true_value), {})
-    _run_vng(ast, _generate_options())
-    assert var._name == "dTmp42"
+    assert assignment.destination.name == "unkVar0"
 
 
 def test_same_variable():
     """Variables can be copies of the same one. The renamer should only rename a variable once. (More times would destroy the actual name)"""
-    true_value = LogicCondition.initialize_true(LogicCondition.generate_new_context())
     var1 = Variable("tmp_42", Float(64))
-    var2 = Variable("var_0", Integer(104, True))
-    ast = AbstractSyntaxTree(
-        CodeNode(
-            [Assignment(var1, Constant(0)), Assignment(var1, Constant(0)), Assignment(var2, Constant(0)), Assignment(var2, Constant(0))],
-            true_value,
-        ),
-        {},
+    node = CodeNode(
+        [Assignment(var1, Constant(0)), Assignment(var1, Constant(0))],
+        LogicCondition.initialize_true(LogicCondition.generate_new_context()),
+    )
+    ast = AbstractSyntaxTree(node, {})
+    _run_vng(ast, _generate_options())
+    assert node.instructions[0].destination.name == "dTmp0"
+    assert node.instructions[1].destination.name == "dTmp0"
+
+
+def test_same_variable_idx():
+    """Variables with the same counter should not be renamed into the same thing"""
+    var1 = Variable("x_1", Integer.int32_t())
+    var2 = Variable("y_1", Integer.int32_t())
+    node = CodeNode(
+        [Assignment(var1, Constant(0)), Assignment(var2, Constant(0))],
+        LogicCondition.initialize_true(LogicCondition.generate_new_context()),
+    )
+    ast = AbstractSyntaxTree(node, {})
+    _run_vng(ast, _generate_options())
+    assert node.instructions[0].destination.name != node.instructions[1].destination.name
+
+
+def test_different_custom_names_0():
```

Comment:
What is with the special variable names `exit_` and `entry_` for multiple exit-\entry-restructuring?

Reply:
Both of them have a size of >2 => `_get_name_identifier` will purge them and return as a base.
=> z.B. dExit0 (if float)

Will add them to this test case.